Extensions/PathfindingBehavior/pathfindingruntimebehavior.ts

```diff
@@ -728,21 +874,42 @@ namespace gdjs {
             (obj.getDrawableY() + obj.getHeight() + this._topBorder) /
               this._cellHeight
           );
-          if (
-            topLeftCellX < xPos &&
+
+          objectsOnCell = topLeftCellX < xPos &&
             xPos < bottomRightCellX &&
             topLeftCellY < yPos &&
-            yPos < bottomRightCellY
-          ) {
-            objectsOnCell = true;
-
-            //The cell is impassable, stop here.
-
-            //Superimpose obstacles
+            yPos < bottomRightCellY;
+          }
+          else {
+            this._aabb.min[0] = nodeCenter[0] - this._leftBorder;
+            this._aabb.min[1] = nodeCenter[1] - this._topBorder;
+            this._aabb.max[0] = nodeCenter[0] + this._rightBorder;
+            this._aabb.max[1] = nodeCenter[1] + this._bottomBorder;
+            this._obstacles.getAllObstaclesAroundAABB(this._aabb, this._closeObstacles);
+            
+            if (this._behavior._collisionMethod == PathfindingRuntimeBehavior.HITBOX_COLLISION) {
+              this._moveStampHitBoxesTo(nodeCenter[0], nodeCenter[1]);
+              objectsOnCell = this._checkCollision(obj.getHitBoxes());
+              
+              obj.getHitBoxes()[0].computeCenter();
```

Comment:
Why is this needed? 
Looks fairly dangerous to manually go ahead and modify the hitboxes of an object (even if it's just calling computeCenter - it's the responsibility of the object to do so. And objects could have multiple hitboxes).

Or is it just for debug?

Reply:
Just for debug, I did a better log